### PR TITLE
Add support for browserless login

### PR DESCRIPTION
**Pull request by joshma**
Instead of opening a browser always, we allow the user to enter a token
from a page `/cli/login?showToken=1`, and only if the user hits ENTER do
we proceed with the normal browser route.

This way, users can sign in if they're e.g. on a remote machine.
 
screenshot_20220103_N4E8QZQi

### CODE REVIEW

File: pkg/api/client.go [hunk 1]
```diff
@@ -100,7 +100,13 @@ func (c Client) appURL() *url.URL {
 	return u
 }
 
-// LoginURL returns a login URL that redirects to `uri`.
+func (c Client) TokenURL() string {
+	u := c.appURL()
+	u.Path = "/cli/login"
+	u.RawQuery = url.Values{"showToken": []string{"1"}}.Encode()
```
Comment (colinking) [inline pkg/api/client.go:106]: Minor, but `/cli/token` would look a bit cleaner imo

**Review by colinking — APPROVED** [review]

File: pkg/cmd/auth/login/login.go [hunk 2]
```diff
@@ -94,6 +96,40 @@ func EnsureLoggedIn(ctx context.Context, c *cli.Config) error {
...
+			cfg.Tokens = map[string]string{}
+		}
+		cfg.Tokens[c.Client.Host] = token
+		if err := conf.WriteDefault(cfg); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	logger.Log("Enter your %s from %s", logger.Bold("token"), logger.Blue(c.Client.TokenURL()))
```
[the remark this answers is not in the document]
Comment (joshma) [reply pkg/cmd/auth/login/login.go:115]: Yep, saw Pulumi do this most recently so I went with that. The bolding is just from `survey` but I'll look into configuring it.

File: pkg/api/client.go (hunk 1 above)
Comment (joshma) [reply pkg/api/client.go:106]: Yeaaah I hear you, but then it's a whole other component and it felt too related to `/login` that I then justified leaving it at the same URL.

**Review by ericliuche — APPROVED** [review]

File: pkg/cmd/auth/login/login.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshma) [reply pkg/cmd/auth/login/login.go:115]: screenshot_20220103_N4E8QZQi

Updated

**Pull request merged**